### PR TITLE
Add coloured badge borders

### PR DESCRIPTION
**Pull request by paulgain**
------------------------
Badges
------------------------
COMPLETE = GREEN_BORDER
UPCOMING = BLUE_BORDER
INCOMPLETE = BLUE_BORDER
CANCELLED = RED_BORDER
COMPLETED_SERVICE_DELIVERY = RED_BORDER

badge-borders

https://trello.com/c/7z7F8Vd4/1296-update-badge-colour-based-upon-status-ie-cancelled-interaction-is-red

### CODE REVIEW

File: src/activity-feed/activity-feed-cards/InteractionUtils.js [hunk 1]
```diff
@@ -17,6 +22,14 @@ const BADGE_LABELS = {
   COMPLETED_SERVICE_DELIVERY: 'Completed service delivery',
 }
 
+const BADGE_BORDERS = {
+  COMPLETE: GREEN_BORDER,
+  UPCOMING: BLUE_BORDER,
+  INCOMPLETE: BLUE_BORDER,
+  CANCELLED: RED_BORDER,
+  COMPLETED_SERVICE_DELIVERY: RED_BORDER,
```
Comment (yeahfro) [inline src/activity-feed/activity-feed-cards/InteractionUtils.js]: Is it right that service delivery is `RED`?

File: src/activity-feed/activity-feed-cards/card/CardMeta.jsx [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import React from 'react'
 import styled from 'styled-components'
 import { SPACING } from '@govuk-react/constants'
-
+import { GREY_2 } from 'govuk-colours'
 import PropTypes from 'prop-types'
 import DateUtils from '../../../utils/DateUtils'
 
+const DEFAULT_BORDER = `2px solid ${GREY_2};`
```
Comment (yeahfro) [inline src/activity-feed/activity-feed-cards/card/CardMeta.jsx]: The extra semi colon is not required here

File: src/activity-feed/activity-feed-cards/card/CardMeta.jsx [hunk 3]
```diff
@@ -23,15 +25,24 @@ export default class CardMeta extends React.Component {
   static propTypes = {
     startTime: PropTypes.string.isRequired,
     badge: PropTypes.string,
```
Comment (yeahfro) [inline src/activity-feed/activity-feed-cards/card/CardMeta.jsx]: `badge` could be a more complex `object` with `text` and `colour` fields

**Comment by paulgain** [comment]
> Looks good. I am thinking that something to consider is a more complex `object` such as:
> 
> ```
> const BADGES = {
>   COMPLETE: {
>     text: 'Completed interaction',
>     colour: GREEN,
>   },
>   UPCOMING: {
>     text: 'Upcoming interaction',
>     colour: BLUE,
>   },
>   INCOMPLETE: {
>     text: 'Incomplete interaction',
>     colour: BLUE,
>   },
>   CANCELLED: {
>     text: 'Cancelled interaction',
>     colour: RED,
>   },
>   COMPLETED_SERVICE_DELIVERY: {
>     text: 'Completed service delivery',
>     colour: RED,
>   },
> }
> ```
> 
> And then the border CSS is not in `InteractionUtils.js`.
> 
> This would imply that the `Badge` styled component is:
> 
> ```
> const Badge = styled('span')`
>   border: ${({ borderColour }) => `2px solid ${borderColour || GREY_2}`};
>   border-radius: 4px;
>   padding: 2px 4px;
> `
> ```

I agree the complex `object` would be better 👍 

I think specific styles like badge `border` belong in the card, that way for each card we can pass in different types of badge border rather than hardcoding the border style and width and just passing in the colour as you suggest.

File: src/activity-feed/activity-feed-cards/card/CardMeta.jsx (hunk 3 above)
Comment (paulgain) [reply src/activity-feed/activity-feed-cards/card/CardMeta.jsx]: 👍

File: src/activity-feed/activity-feed-cards/card/CardMeta.jsx (hunk 2 above)
Comment (paulgain) [reply src/activity-feed/activity-feed-cards/card/CardMeta.jsx]: Doh!

File: src/activity-feed/activity-feed-cards/InteractionUtils.js [hunk 4]
```diff
@@ -17,6 +22,14 @@ const BADGE_LABELS = {
   COMPLETED_SERVICE_DELIVERY: 'Completed service delivery',
 }
 
+const BADGE_BORDERS = {
```
Comment (rafenden) [inline src/activity-feed/activity-feed-cards/InteractionUtils.js]: I believe there should be a `CardBadge` component which owns this code.

Comment (paulgain) [reply src/activity-feed/activity-feed-cards/InteractionUtils.js]: We already have that here: 

https://github.com/uktrade/data-hub-components/blob/4530d0ae54a0b5affd54d31b0204042d8587d7f3/src/activity-feed/activity-feed-cards/card/CardMeta.jsx#L40-L44

The badge data is simply passed into the component via `props`.

Comment (yeahfro) [reply src/activity-feed/activity-feed-cards/InteractionUtils.js]: @rafenden this is good point which we have just been talking about. We need to remove `InteractionUtils` as it's sending this in a bad direction. This has been captured in a separate ticket which @paulgain is keen to follow up on. https://trello.com/c/vYo85TDr/1349-refactor-interactionutils-remove-it

**Pull request merged**